### PR TITLE
Warn for deprecations on ``dask.config.set()``

### PR DESCRIPTION
**Pull request by crusaderky**
`dask.config.set` on distributed-specific deprecated keys will now log a warning.

- New tests need https://github.com/dask/dask/pull/10499 to pass
- Also fixes failure in already existing test introduced by the above PR

### CODE REVIEW

**Comment by crusaderky** [comment]
@jrbourbeau ready for review (but blocked by dask PR)

File: distributed/config.py [hunk 1]
```diff
@@ -55,7 +55,11 @@
     "rmm": "distributed.rmm",
 }
 
-dask.config.rename(aliases)
+# Affects yaml and env variables configs, as well as calls tgo dask.config.set()
```
Comment (jrbourbeau) [inline distributed/config.py:58]:
```suggestion
# Affects yaml and env variables configs, as well as calls to dask.config.set()
```

**Review by jrbourbeau — APPROVED** [review]
Thanks @crusaderky

**Comment by jrbourbeau** [comment]
Rerunning CI with caching off to make sure tests here pass

**Pull request merged**